File: 179.js
// LeetCode 179 largest Number
// Reference: https://leetcode.com/problems/largest-number/?envType=featured-list&envId=top-interview-questions?envType=featured-list&envId=top-interview-questions

/**
 * @param {number[]} nums
 * @return {string}
 */
var largestNumber = function(nums) {

    if ( !nums || nums.length === 0){
        return '0';
    }

    nums.sort((a, b) => `${b}${a}` - `${a}${b}`);

    if(nums[0] === 0){
        return '0';
    }

    return nums.join('');
    
};

File: 179.py
# LeetCode 179 largest Number
# Reference:https://www.youtube.com/watch?v=WDx6Y4i4xJ8

class Solution:
    def largestNUmber(self, nums: List[int]) -> str:

        for i, n in enumerate(nums):
            nums[i] = str(n)
        
        def compare(n1, n2):
            if n1 + n2 > n2 + n1:
                return -1
            else:
                return 1
        nums = sorted(nums, key=cmp_to_key(compare))
        return str(int("".join(nums)))